doc/source/developing/testing.rst

```diff
@@ -148,358 +106,292 @@ functionality.
+When adding tests to an existing set of answers (like ``local_owls_000.yaml`` or ``local_varia_000.yaml``),
 it is considered best practice to first submit a pull request adding the tests WITHOUT incrementing
 the version number. Then, allow the tests to run (resulting in "no old answer" errors for the missing
 answers). If no other failures are present, you can then increment the version number to regenerate
 the answers. This way, we can avoid accidentally covering up test breakages.
 
-Adding New Answer Tests
-~~~~~~~~~~~~~~~~~~~~~~~
+.. _handling_dependencies:
 
-In order to add a new set of answer tests, it is sufficient to extend the
-*answer_tests* list in *tests/tests.yaml* e.g.:
+Handling yt dependencies
+------------------------
 
-.. code-block:: diff
+We attempt to make yt compatible with a wide variety of upstream software
+versions. However, sometimes a specific version of a project that yt depends on
+causes some breakage and must be blacklisted in the tests or a more
+experimental project that yt depends on optionally might change sufficiently
+that the yt community decides not to support an old version of that project.
 
-   --- a/tests/tests.yaml
-   +++ b/tests/tests.yaml
-   @@ -60,6 +60,10 @@
-        - yt/analysis_modules/absorption_spectrum/tests/test_absorption_spectrum.py:test_absorption_spectrum_non_cosmo
-        - yt/analysis_modules/absorption_spectrum/tests/test_absorption_spectrum.py:test_absorption_spectrum_cosmo
+To handle cases like this, the versions of upstream software projects installed
+on the machines running the yt test suite are pinned to specific version
+numbers that must be updated manually. This prevents breaking the yt tests when
+a new version of an upstream dependency is released and allows us to manage
+updates in upstream projects at our pace.
+
+If you would like to add a new dependency for yt (even an optional dependency)
+or would like to update a version of a yt dependency, you must edit the
+``tests/test_requirements.txt`` file, this path is relative to the root of the
+repository. This file contains an enumerated list of direct dependencies and
+pinned version numbers. For new dependencies, simply append the name of the new
+dependency to the end of the file, along with a pin to the latest version
+number of the package. To update a package's version, simply update the version
+number in the entry for that package.
 
-   +  local_gdf_000:
-   +    - yt/frontends/gdf/tests/test_outputs.py
-   +
-   +
-    other_tests:
-      unittests:
+Finally, we also run a set of tests with "minimal" dependencies installed. Please make sure any new tests you add that depend on an optional dependency are properly set up so that the test is not run if the dependency is not installed. If for some reason you need to update the listing of packages that are installed for the "minimal" dependency tests, you will need to edit ``tests/test_minimal_requirements.txt``.
```

Comment:
Not necessarily pytest related but I think it would be helpful to explain the following sentence a bit more: "Please make sure any new tests you add that depend on an optional dependency are properly set up." Maybe replace with something like:

"When adding tests that depend on an optional dependency, you can wrap the test with the `yt.testing.requires_module` decorator to ensure it does not run during the minimial dependency tests (see `yt/frontends/amrvac/tests/test_read_amrvac_namelist.py` for a good example)."   

(note: I'm not actually sure if this is the way to "properly set up". maybe there are other better ways? maybe it's changed with pytest? but it'd be nice to explain a little bit of what "properly set up" means.)